### PR TITLE
Add syntax rule for DistSQL

### PR DESCRIPTION
**Pull request by Mike0601**
For #18811, #19942

### CODE REVIEW

File: docs/document/content/user-manual/shardingsphere-proxy/distsql/_index.cn.md [hunk 1]
```diff
@@ -4,4 +4,31 @@ weight = 3
...
+- 表名
+- 列名
+- 索引名称
+- 资源名称
+- 规则名称
+- 算法名称
+
+1. 标识符中允许使用的字符有：[a-z,A-Z,0-9,_]（字母、数字、下划线），且应以字母开头。
+1. 当标识符中出现关键字或特殊字符时，使用反引号 (`) 。
+### 字面量
```
Comment (terrymanu) [inline docs/document/content/user-manual/shardingsphere-proxy/distsql/_index.cn.md:27]: Please leave a blank line between new title and last content.

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: docs/document/content/user-manual/shardingsphere-proxy/distsql/_index.en.md [hunk 2]
```diff
@@ -4,4 +4,32 @@ weight = 3
...
+
+1. When keywords or special characters appear in the identifier, use the backticks (`).
+
+### Literal
+
+Types of literals include:
+
+- string: enclosed in single quotes (') or double quotes (")
+- int: it is generally a positive integer, such as 0-9;
+Note: some DistSQL syntax allows negative values. In this case, a negative sign (-) can be added before the number, such as - 1.
```
Comment (RaigorJiang) [inline docs/document/content/user-manual/shardingsphere-proxy/distsql/_index.en.md:34]: No space required between minus and number.

Comment (Mike0601) [reply docs/document/content/user-manual/shardingsphere-proxy/distsql/_index.en.md]: have removed space.

File: docs/document/content/user-manual/shardingsphere-proxy/distsql/_index.cn.md (hunk 1 above)
Comment (Mike0601) [reply docs/document/content/user-manual/shardingsphere-proxy/distsql/_index.cn.md:29]: have added a blank line.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**